components/TextFields/src/MDCTextInputControllerLegacyDefault.m

```diff
@@ -218,22 +221,31 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
           decodeIntegerForKey:MDCTextInputControllerLegacyDefaultCharacterCountViewModeKey];
     }
     _disabledColor =
-        [aDecoder decodeObjectForKey:MDCTextInputControllerLegacyDefaultDisabledColorKey];
-    _errorColor = [aDecoder decodeObjectForKey:MDCTextInputControllerLegacyDefaultErrorColorKey];
+        [aDecoder decodeObjectOfClass:[UIColor class]
+                               forKey:MDCTextInputControllerLegacyDefaultDisabledColorKey];
+    _errorColor = [aDecoder decodeObjectOfClass:[UIColor class]
+                                         forKey:MDCTextInputControllerLegacyDefaultErrorColorKey];
     _floatingEnabled =
         [aDecoder decodeBoolForKey:MDCTextInputControllerLegacyDefaultFloatingEnabledKey];
     _floatingPlaceholderNormalColor = [aDecoder
-        decodeObjectForKey:MDCTextInputControllerLegacyDefaultFloatingPlaceholderNormalColorKey];
+        decodeObjectOfClass:[UIColor class]
+                     forKey:MDCTextInputControllerLegacyDefaultFloatingPlaceholderNormalColorKey];
     _floatingPlaceholderScale = [aDecoder
-        decodeObjectForKey:MDCTextInputControllerLegacyDefaultFloatingPlaceholderScaleKey];
+        decodeObjectOfClass:[NSNumber class]
+                     forKey:MDCTextInputControllerLegacyDefaultFloatingPlaceholderScaleKey];
     _inlinePlaceholderColor =
-        [aDecoder decodeObjectForKey:MDCTextInputControllerLegacyDefaultInlinePlaceholderColorKey];
+        [aDecoder decodeObjectOfClass:[UIColor class]
+                               forKey:MDCTextInputControllerLegacyDefaultInlinePlaceholderColorKey];
     _leadingUnderlineLabelTextColor = [aDecoder
-        decodeObjectForKey:MDCTextInputControllerLegacyDefaultLeadingUnderlineLabelTextColor];
-    _normalColor = [aDecoder decodeObjectForKey:MDCTextInputControllerLegacyDefaultNormalColorKey];
-    _textInput = [aDecoder decodeObjectForKey:MDCTextInputControllerLegacyDefaultTextInputKey];
+        decodeObjectOfClass:[UIColor class]
+                     forKey:MDCTextInputControllerLegacyDefaultFloatingPlaceholderNormalColorKey];
```

Comment:
This needs to be `MDCTextInputControllerLegacyDefaultLeadingUnderlineLabelTextColor`